src/RecursiveApply/RecursiveApply.jl

```diff
@@ -15,12 +15,7 @@ export ⊞, ⊠, ⊟, tuplemap
 A simpler `map` impl for mapping function `fn` a tuple argument `tup`
 """
 @inline function tuplemap(fn::F, tup::Tuple) where {F}
-    N = length(tup)
-    ntuple(Val(N)) do I
-        Base.@_inline_meta
-        @inbounds elem = tup[I]
-        fn(elem)
-    end
+    map(fn, tup)
```

Comment:
@jakebolewski, is there a similar change we can/should make below? I don't see it in the allocation plots, so maybe it's not worth it?

Reply:
should we just get rid of `tuplemap` then?

Reply:
We could, but `tuplemap` is still showing up as a leading allocation contributor in one case (shallow water barotropic instability), so it might be easier to leave it in case we need to try hacking it again. I haven't (yet) dug into who is calling it for that case, so I'm not sure what the cause is.

Reply:
It would be nice to keep it factored out as in the future I suspect we will have to undo some of the tuple operations

Reply:
It is used in so many places if we ever have to change it from map to something more optimized we can.